LabExT/Movement/PathPlanning.py

```diff
@@ -48,6 +49,55 @@ class StagePolygon(ABC):
     This class cannot be initialised.
     """
 
+    @classmethod
+    def load(cls, polygon_data: dict) -> Type[StagePolygon]:
+        """
+        Returns a stage polygon reconstructed from polygon data.
+        """
+        polygon_name = polygon_data.get("polygon_cls")
+        if not polygon_name:
+            raise ValueError(
+                "Cannot find and load stage polygon without polygon class name. "
+                "Please provide a key 'polygon_cls' in polygon data.")
+
+        if (isinstance(polygon_name, type) and
+                issubclass(polygon_name, StagePolygon)):
+            polygon_cls = polygon_name
+        elif isinstance(polygon_name, str):
+            available_classes = StagePolygon.find_polygon_classes()
+            try:
+                polygon_cls = next(
+                    pg for pg in available_classes if pg.__name__ == polygon_name)
+            except StopIteration:
+                available_classes_str = ", ".join(
+                    map(lambda pg: pg.__name__, available_classes))
+                raise ValueError(
+                    f"No polygon class found with name {polygon_name}. The following are available: "
+                    f"{available_classes_str}")
+        else:
+            raise ValueError(
+                f"Cannot restore stage polygon class with 'polygon_cls' equal to {polygon_name} of type {type(polygon_name)}. "
+                "Please provide a value for 'polygon_cls' of type str or StagePolygon. "
+                "If you provide a string, it must be the name of the polygon class.")
+
+        try:
+            orientation = Orientation[polygon_data["orientation"]]
+        except KeyError as err:
+            raise ValueError(
+                f"The parameter is not defined: {err}. "
+                "Make sure to pass a valid orientation.")
```

Comment:
try to be more explicit here. What parameter? How do I get a valid orientation?

Reply:
Updated!